### ISSUE TITLE
[HorizontalPodAutoscaler Walkthrough] Metrics Server can't seem to be applied properly

### ISSUE THREAD

**Issue by ZCai319** [kubernetes/website#33991]
I tried to autoscale my k3d cluster. Followed this tutorial and it told me to apply this metrics server yaml. Used command: 
`kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml`

Response I get from terminal is:
```
Warning: resource serviceaccounts/metrics-server is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. 
kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. 
The missing annotation will be patched automatically.
serviceaccount/metrics-server configured

Warning: resource clusterroles/system:aggregated-metrics-reader is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. 
kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. 
The missing annotation will be patched automatically.
clusterrole.rbac.authorization.k8s.io/system:aggregated-metrics-reader configured

Warning: resource clusterroles/system:metrics-server is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. 
kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. 
The missing annotation will be patched automatically.
clusterrole.rbac.authorization.k8s.io/system:metrics-server configured

Warning: resource rolebindings/metrics-server-auth-reader is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. 
kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. 
The missing annotation will be patched automatically.
rolebinding.rbac.authorization.k8s.io/metrics-server-auth-reader configured

Warning: resource clusterrolebindings/metrics-server:system:auth-delegator is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. 
kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. 
The missing annotation will be patched automatically.
clusterrolebinding.rbac.authorization.k8s.io/metrics-server:system:auth-delegator configured

Warning: resource clusterrolebindings/system:metrics-server is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply
 should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
clusterrolebinding.rbac.authorization.k8s.io/system:metrics-server configured

Warning: resource services/metrics-server is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be us
ed on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
service/metrics-server configured

Warning: resource deployments/metrics-server is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be
 used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
deployment.apps/metrics-server configured

Warning: resource apiservices/v1beta1.metrics.k8s.io is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should
 only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
apiservice.apiregistration.k8s.io/v1beta1.metrics.k8s.io configured
```
Wasn't sure what it was, but it said it'll be patched automatically so alas. I applied my yml files for the k3d cluster so the services (that worked without scaling) can run on my cluster. I than proceeded to use these commands:
```
kubectl autoscale deployment gateway-service-d --cpu-percent=50 --min=1 --max=10
kubectl autoscale deployment user-service-d --cpu-percent=50 --min=1 --max=10
kubectl autoscale deployment kweet-service-d --cpu-percent=50 --min=1 --max=10
kubectl get hpa
```

And the response I get back is:
```
NAME                REFERENCE                                                       TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
gateway-service-d   Deployment/gateway-service-d                                    <unknown>/50%   1         10        1          15m
user-service-d      Deployment/user-service-d                                       <unknown>/50%   1         10        1          15m
kweet-service-d     Deployment/kweet-service-d                                      <unknown>/50%   1         10        1          15m
hpa                 Deployment/gateway-service-d, kweet-service-d, user-service-d   <unknown>/10%   1         10        0          97s

```

It must have something to do with the previous failed application of yml file, but I can't find anything on how to fix it.

**Comment by sftim** [comment]
What version of Kubernetes are you running @ZCai319, and what page are you reading to follow the steps from (can you let us know a URL?)

**Comment by ZCai319** [comment]
Following this walkthrough: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/. I'm using GitVersion:"v1.23.5".

 What I realized is that it actually **did** work. I had to `kubectl get hpa --watch` then it showed me that the gateway was working fine (instead of unknown it said 0%), but the other 2 weren't. So an acquaintance told me to restart my laptop. I did, reran the cluster, `kubectl get hpa --watch` and it started working.

Kind of annoying though, that I have to restart my laptop every time for it to work.
